compiler/noirc_frontend/src/hir/resolution/errors.rs

```diff
@@ -150,6 +150,8 @@ pub enum ResolverError {
     AttributeFunctionIsNotAPath { function: String, span: Span },
     #[error("Attribute function `{name}` is not in scope")]
     AttributeFunctionNotInScope { name: String, span: Span },
+    #[error("The trait `{the_trait}` is not implemented for `{typ}")]
+    TraitNotImplemented { the_trait: String, typ: String, span: Span },
```

Comment:
This is only issued for child trait requirements, right? Perhaps we should be more explicit why the trait needs to be implemented. E.g. rust says:
```
error[E0277]: the trait bound `S: Foo` is not satisfied
  --> foo.rs:10:14
   |
10 | impl Bar for S {}
   |              ^ the trait `Foo` is not implemented for `S`
   |
   = help: the trait `Foo` is implemented for `()`
note: required by a bound in `Bar`
  --> foo.rs:3:13
   |
3  | trait Bar : Foo {}
   |             ^^^ required by this bound in `Bar`
```

Reply:
It seems we don't have a way for an error to have two locations? I see `Diagnostic::simple_error` has "secondary span" but that seems to be the primary span (or at least I can't find a primary span)

Reply:
Hm, maybe there's a way to do it without those convenience methods... I'll check.

Reply:
I think I got it! Here's what it looks like now:

```
error: The trait bound `Struct: Foo` is not satisfied
  ┌─ src/main.nr:3:16
  │
3 │ pub trait Bar: Foo {}
  │                --- required by this bound in `Bar
  ·
7 │ impl Bar for Struct {}
  │              ------ The trait `Foo` is not implemented for `Struct
```